server/src/main/java/org/opensearch/action/search/SearchPhaseController.java

```diff
@@ -636,7 +636,7 @@ private static Sort createSort(TopFieldDocs[] topFieldDocs) {
      */
     private static boolean isSortWideningRequired(TopFieldDocs[] topFieldDocs, int sortFieldindex) {
         for (int i = 0; i < topFieldDocs.length - 1; i++) {
-            if (topFieldDocs[i].fields[sortFieldindex] != topFieldDocs[i + 1].fields[sortFieldindex]) {
+            if (!topFieldDocs[i].fields[sortFieldindex].equals(topFieldDocs[i + 1].fields[sortFieldindex])) {
```

Comment:
@gashutos mind taking a look? I found this one out while adding tests for multi index search queries

Reply:
Oh, I remember I did test this with int & long and it worked fine. 
The unit tests too are working fine, really is this the issue ?

Reply:
Oh got it, so correctness wise it was still giving right results ?

Reply:
Yes, no visible effect

Reply:
Was this same test passing with '!=' operator ?

Reply:
yea, its good to have. I was just curious why it is failing :)
>you mean we don't need this test 260_sort_mixed.yml
No no, we need it, thank you for adding it...

Reply:
@reta I think this is mixed cluster scenario which is failing in ```./gradlew test```.
Our old versions dont support mixig results between LONG/DOUBLE.  We are specifying sortType Long for Long NumericType [here](https://github.com/gashutos/OpenSearch/blob/5cf1e759b0b076de2c7c50b13d3b9a0e1c87c3e9/server/src/main/java/org/opensearch/index/fielddata/IndexNumericFieldData.java#L73), and SortType Double for Double numeric type. So it will return results in Long for those shards of index with Long mapping, while Double for another case.

The new changes we did as part of widening sort type handles it correctly but that will be in only latest version.
May be we can change mapping types from ```Long to Double``` to ```Int to Long``` or ```Float to Double``` to having them work in mixed cluster scenario.

Reply:
> The new changes we did as part of widening sort type handles it correctly but that will be in only latest version.

Correct, so I was adding the version check here:

```
  - skip:
      version: " - 2.6.99"
      reason: relies on numeric sort optimization that landed in 2.7.0 only
```

So the test should be skipped for anything below 2.7.0 (we backported this change to 2.x so it should work for 3.0.0 and 2.7.0) but for some reasons, the test is run on versions it is not supposed to  ... I am curious to find out why ....